### DISCUSSION TITLE
On clustered engines

### DISCUSSION THREAD

**Discussion by bohledevs** [StephenOTT/Camunda-Monitoring#18] [Q&A] [answered]
Hello there Stephen,

Thank you for this amazing contribution.

I was wondering if your plugin would be applicable to a clustered engine, since I had some consistency issues trying to apply it to multiple Camunda Spring-Boot nodes.

I know it works perfectly on Camunda Run. Just trying to find a workaround to take it to a clustered architecture.

Best Regards,
                        Facundo Bohle

**Comment by StephenOTT** [comment, marked as answer]
You only need to run the plugin on 1 node.  Think of the Monitoring plugin as a dedicated node (at scale / in larger cluster).
This is why the sample code has the Job Executor disabled: https://github.com/StephenOTT/Camunda-Monitoring#camunda-run-quick-start-with-prometheus.  It is a node which never does anything for the overall engine except for running monitoring queries.

**Reply by bohledevs** [reply]
This really clarified my understanding of the plugin. 
I was attaching them to every node in the cluster and it was producing incorrect data. This was a mistaken notion I've been carrying since your previous plugin.

Thank you again!

**Reply by StephenOTT** [reply]
The one area where you may want to run the plug-in on every node is if you are counting occurrences on the specific node: such as intercepting a Command, counting durations of executing scripts or other runtime specific actions: you would count these in memory as they occur and then scrape the data across all nodes
